services/notification_listener.go

```diff
@@ -91,14 +100,26 @@ func (nl *NotificationListener) AddJob(job models.Job) error {
 		return nil
 	}
 
-	sub, err := nl.Store.TxManager.SubscribeToLogs(nl.logNotifications, addresses)
+	sub, err := nl.Store.TxManager.SubscribeToLogs(nl.logNotifications, nl.filterQueryFor(addresses))
 	if err != nil {
 		return err
 	}
 	nl.addSubscription(sub)
 	return nil
 }
 
+func (nl *NotificationListener) filterQueryFor(addresses []common.Address) ethereum.FilterQuery {
+	blockHeader := nl.HeadTracker.Get()
+	var fromBlock *big.Int
+	if blockHeader != nil {
+		fromBlock = blockHeader.Number.ToInt()
+	}
+	return ethereum.FilterQuery{
+		FromBlock: fromBlock,
```

Comment:
For the future: We aren't keeping track of the block hash now, but when we'll probably need to go back a few blocks from our past head and make sure the last few blocks still match up.